### PR TITLE
Update Developer Guide with Logic component

### LINKED ISSUE #157: Update Developer Guide with Logic component
The Logic section of the Developer Guide is incomplete.

### PR DESCRIPTION
**Pull request by petermonky**
Closes #157.

Updates the Logic section of the developer guide as well as relevant UML diagrams.

### CODE REVIEW

**Comment by BrendonLau** [comment]
LGTM

File: docs/diagrams/LogicClassDiagram.puml [hunk 1]
```diff
@@ -27,8 +27,8 @@ Class HiddenOutside #FFFFFF
 HiddenOutside ..> Logic
 
 LogicManager .right.|> Logic
-LogicManager -right->"1" AddressBookParser
-AddressBookParser ..> XYZCommand : creates >
+LogicManager -right->"1" CompanyListParser
+CompanyListParser ..> XYZCommand : creates >
```
Comment (kerbysoh) [inline docs/diagrams/LogicClassDiagram.puml:32]: Not necessary but we can consider having a high level overview of parser and command class interaction within logic, and then breakdown the parser and command classes individually using two separate diagrams.

E.g. Can refer to the Model classes for reference

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -68,11 +68,34 @@ The `UI` component,
...
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteCompanyCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+The other classes in `Logic` (omitted from the class diagram above) that are used for parsing a user command can be represented as follows:
+
+<img src="images/ParserClasses.png" width="600"/>
+
+How the parsing works:
+
+* When called upon to parse a user command, the `CompanyListParser` class creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCompanyCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCompanyCommand`) which the `CompanyListParser` returns back as a `Command` object.
```
Comment (kerbysoh) [inline docs/DeveloperGuide.md]: Possibly specify which classes Parser uses to Parse for more clarity

**Review by kerbysoh — COMMENTED** [review]
Explanation is very detailed and accurate. Consider the minor changes.

File: docs/diagrams/LogicClassDiagram.puml (hunk 1 above)
Comment (petermonky) [reply docs/diagrams/LogicClassDiagram.puml:32]: Okay this sounds good 👍 I'll try to write something up for this.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (petermonky) [reply docs/DeveloperGuide.md]: That would definitely help 👍 thanks!

**Pull request merged**